modules/serverless-kafka-source-kn.adoc

```diff
@@ -0,0 +1,103 @@
+//
+// * serverless/event_sources/serverless-kafka-source.adoc
+[id="serverless-kafka-source-kn_{context}"]
+= Creating a Kafka event source using the `kn` CLI
+
+This section describes how to create a KafkaSource using the `kn` command.
+
+:FeatureName: Creating a Kafka event source using the `kn` CLI
+include::../modules/technology-preview.adoc[leveloffset=+0]
+
+.Prerequisites
+
+* The {ServerlessOperatorName}, Knative Eventing, Knative Serving, and the `KnativeKafka` custom resource definitions are installed on your cluster.
+* You have created a project or have access to a project with the appropriate roles and permissions to create applications and other workloads in {product-title}.
+* You have access to a Kafka cluster that produces the Kafka messages you want to import.
+
+.Procedure
+
+. To verify that the Kafka source is working, create a Knative service that dumps incoming messages into the service logs:
++
+[source, terminal]
+----
+$ kn service create event-display \
+    --image quay.io/openshift-knative/knative-eventing-sources-event-display:latest
+----
+
+. Create a `KafkaSource` resource:
++
+[source,terminal]
+----
+$ kn source kafka create mykafkasrc \
+    --servers my-cluster-kafka-bootstrap.kafka.svc:9092 \
+    --topics test-topic --consumergroup test-consumer-group \
+    --sink event-display
+----
++
+The `--servers`, `--topics`, and `--consumergroup` options specify the connection parameters to the Kafka cluster. The `--consumergroup` option is optional.
++
+Additionally, this command makes several assumptions, including:
++
+* `namespace` is `kafka`
+* `pod` is `my-cluster-kafka-0`
+
+. Optional. View details about the created Kafka source:
++
+[source, terminal]
+----
+$ kn source kafka describe mykafkasrc
+Name:              mykafkasrc
+Namespace:         default
+Age:               1h
+BootstrapServers:  my-cluster-kafka-bootstrap.kafka.svc:9092
+Topics:            test-topic
+ConsumerGroup:     test-consumer-group
+
+Sink:
+  Name:       event-display
+  Namespace:  default
+  Resource:   Service (serving.knative.dev/v1)
+
+Conditions:
+  OK TYPE            AGE REASON
+  ++ Ready            1h
+  ++ Deployed         1h
+  ++ SinkProvided     1h
+----
+
+.Verification steps
+
+. Trigger the Kafka instance to send a message to the topic:
++
+[source,terminal]
+----
+$ oc -n kafka exec my-cluster-kafka-0 \
+    -ti -- bin/kafka-console-producer.sh \
+    --bootstrap-server localhost:9092 \
+    --topic test-topic
+----
++
+Enter the message in the prompt.
++
+This command makes several assumptions, including:
++
+* `namespace` is `kafka`
+* `pod` is `my-cluster-kafka-0`
++
+Alternatively, you can send a message without Kafka running in the cluster:
++
+[source,terminal]
+----
+$ oc -n kafka run kafka-producer \
+    -ti --image=quay.io/strimzi/latest-kafka-2.7.0 --rm=true \
+    --restart=Never -- bin/kafka-console-producer.sh \
+    --broker-list my-cluster-kafka-bootstrap:9092 --topic my-topic
+----
+
```

Comment:
Personally, I would only offer one of the two possibilities. After all, its "only" about checking the setup, not the core functionality so having that part blowed up (there are even more possibilities) dilutes the original focus of describing the KafkaSource itself.

I would pick this example here only, and you can omitt the `-n` declaration here if using a full broker domain. So you just would have to mention, that you assume that Kafka is installed in the `kafka` namespace (like above), and then offer this command:

```
$ oc run kafka-producer \
     -ti --image=quay.io/strimzi/latest-kafka-2.7.0 --rm=true \
    --restart=Never -- bin/kafka-console-producer.sh \
    --broker-list my-cluster-kafka-bootstrap.kafka.svc:9092 --topic my-topic
```

You also mention that the `my-topic` needs to match the topic for which you have configured the KafkaSource.

Reply:
Modified the section to only feature this command.

Reply:
About `my-topic`: should it be changed to `test-topic` as in the earlier listings?

Reply:
Probably I've introduced `test-topic` from my example/snippets. However, I like a bit more to keep `my-*` naming to align with `my-cluster-kafka, mykafkasrc`. Wdyt?

Reply:
I like using this "my-" kind of prefix for every user provided data, as it makes clear that this is not something that this is not a keyword or e.g. a fixed namespace (like in "default"). Could by also 'my-test-topic' but 'my-topic' sound nicer imo ;-). 

We should align those and might agree to use a "my" prefix for all this kind of user-given an user "invented" information.

Reply:
@dsimansk @rhuss Did the `s/test-topic/my-topic` and `s/test-consumer-group/my-consumer-group` change across the section.

I've now implemented your feedback and Ashleigh's, could you do another review before I send it to the QE and the final docs team review? Thanks!